doc/source/dev/contributor/debugging_linalg_issues.rst

```diff
@@ -0,0 +1,440 @@
+
+       % otool -L build/scipy/linalg/_fblas.cpython-310-darwin.so
+       build/scipy/linalg/_fblas.*.so:
+           /usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1336.61.1)
+           @rpath/libopenblas.0.dylib (compatibility version 0.0.0, current version 0.0.0)
+
+
+2. Checking whether the linked library contains a given symbol. E.g.,
+   conda-forge installs a ``libblas.so`` that may be any supported library:
+
+.. tab-set::
+
+  .. tab-item:: Linux
+    :sync: linux
+
+    ::
+
+        $ nm -gD ~/mambaforge/envs/scipy-dev/lib/libblas.so | rg openblas_set_num_threads
+        0000000000362990 T openblas_set_num_threads
+
+  .. tab-item:: macOS
+    :sync: macos
+
+    ::
+
+        % nm ~/mambaforge/envs/scipy-dev/lib/libblas.3.dylib | rg openblas_set_num_threads
+        000000000015b6b0 T _openblas_set_num_threads
+
+
+Setting up your environment for switching BLAS libraries
+--------------------------------------------------------
+
+We'll cover several ways of switching between different BLAS libraries, because
+the easiest method may depend on your OS/distro and on whether you want a
+released version of SciPy or a development build.
+
+Conda-forge
+```````````
+Perhaps the easiest way is to use the runtime switching abilities provided by
+distros. For example, to create a conda environment with the latest SciPy
+release installed and then switching between OpenBLAS, Netlib BLAS/LAPACK, and
+MKL is as simple as::
+
+    $ mamba create -n blas-switch scipy threadpoolctl
+    $ mamba activate blas-switch
+    $ python -m threadpoolctl -i scipy.linalg
+    ...
+        "user_api": "blas",
+        "internal_api": "openblas",
+
+    $ mamba install "libblas=*=*netlib"
+    ...
+      libblas                         3.9.0-21_linux64_openblas --> 3.9.0-5_h92ddd45_netlib
+      libcblas                        3.9.0-21_linux64_openblas --> 3.9.0-5_h92ddd45_netlib
+      liblapack                       3.9.0-21_linux64_openblas --> 3.9.0-5_h92ddd45_netlib
+    
+    $ mamba install "libblas=*=*mkl"
+    ...
+      libblas                           3.9.0-5_h92ddd45_netlib --> 3.9.0-21_linux64_mkl
+      libcblas                          3.9.0-5_h92ddd45_netlib --> 3.9.0-21_linux64_mkl
+      liblapack                         3.9.0-5_h92ddd45_netlib --> 3.9.0-21_linux64_mkl
+
+    $ python -m threadpoolctl -i scipy.linalg
+    ...
+      "user_api": "blas",
+      "internal_api": "mkl",
+
+This can be done for development builds as well, when building via ``dev.py``
+in the exact same way as in `SciPy's conda-forge build recipe
+<https://github.com/conda-forge/scipy-feedstock/blob/main/recipe/build.sh>`__
+(outputs omitted for brevity, they're similar to the ones above)::
+
+    $ mamba env create -f environment.yml
+    $ mamba activate scipy-dev
+    $ mamba install "libblas=*=*netlib"  # necessary, we need to build against blas/lapack
+    $ python dev.py build -C-Dblas=blas -C-Dlapack=lapack -C-Duse-g77-abi=true
```

Comment:
It's needed for both MKL and Accelerate. It may be needed for other libraries too, since this is how they're built within conda-forge. But whether it matters will depend on the BLAS/LAPACK function being tested.

Reply:
This one I'd like to keep unchanged; building exactly the way it's done in conda-forge will be most robust.